### PR TITLE
Sync scotch recipe with ptscotch

### PR DESCRIPTION
**Pull request by minrk**
- get meta.yaml, build.sh, Makefile.inc from #16
- move flags configuration to build.sh from Makefile

### CODE REVIEW

File: recipe/Makefile.inc [hunk 1]
```diff
@@ -8,8 +8,8 @@ AR          = ar
 ARFLAGS     = -ruv
 CAT         = cat
 CCS         = ${CC}
-CCP         = ${CCS}
```
Comment (dalcinl) [inline recipe/Makefile.inc]: I dont' like this. For ptscoch, both `CCP` and `CCD` should be `mpicc`. 

Maybe we can do this here:
```
CCS ?= ${CC}
CCP ?= ${CCS}
CCD ?= ${CCP}
```
Then, at `build.sh` in the ptscotch branch, just add `export CCP=mpicc`.

**Comment by dalcinl** [comment]
@minrk At this point I'm wondering whether it would be better to have a separate repo for ptscotch. We will have to manually keep master and ptscotch branches in sync by hand, and the ptscotch branch is going to miss the automatic maintenance from the conda-forge-admin bot. What would be the problem of tracking the ptscotch package in a separate feedstock repo?

**Comment by minrk** [comment]
I don't think it makes a huge difference, but I think it would be easier to keep two branches in sync than two repos, when all the sync work is still the same.

**Comment by minrk** [comment]
Though most of the motivation for ptscotch to be a branch here is because it has been working as a *variant* of scotch, rather than a downstream package. If we are making ptscotch a separate package that *depends on* scotch, then a separate repo starts to make more sense.

If we want to go that route, we should:

1. remove the one-recipe-for-both logic here
2. restart your ptscotch feedstock PR to staged-recipes in a way that depends on scotch and doesn't rebuild it

@basnijholt @ocefpaf opinions on whether ptscotch should get its own repo if it's going to depend on scotch instead of provide a variant of it?

**Comment by dalcinl** [comment]
@minrk Please merge this one and then I'll update the ptscotch branch to get a fresh build. I really need them to get the new PETSc and petsc4py releases out.

Later we can continue the discussion about the separate repo for ptscotch.

**Comment by ocefpaf** [comment]
> @basnijholt @ocefpaf opinions on whether ptscotch should get its own repo if it's going to depend on scotch instead of provide a variant of it?

I don't have a strong opinion. I guess you should go with what is more convenient to the maintainers.

**Comment by minrk** [comment]
@dalcinl merged. Feel free to re-issue the ptscotch PR and I'll merge it.

**Comment by dalcinl** [comment]
@minrk You forgot to `git rm recipe/Makefile.inc.*`.

**Pull request merged**